### PR TITLE
Add io.github.fm_elpac.pmim_ibus

### PR DESCRIPTION
**Pull request by secext2022**
### Please confirm your submission meets all the criteria

Please describe your application briefly below.

PMIM: A Chinese pinyin input method for ibus.

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. 
- [x] All assets referenced in the manifest are redistributable by any party.  
- [x] I am an author/developer/upstream contributor of the project. 
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. 

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: so aarch64 are denied this? I thought Electron was cross platform.

Disappointing

Comment (secext2022) [reply flathub.json:2]: Currently I do not have a machine to actually test this app for aarch64.
Only allow build for aarch64 is still ok ?

Comment (secext2022) [reply flathub.json:2]: There is a binary dependency (librush) currently only tested under x86_64, so it may not work under aarch64.

**Comment by secext2022** [comment]
Add build for aarch64.  But test under aarch64 is not possible.

**Comment by secext2022** [comment]
bot, build io.github.fm_elpac.pmim_ibus

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64", "aarch64"]
```
Comment (bbhtt) [inline flathub.json:2]: These are the only arches available, you can drop this.

File: io.github.fm_elpac.pmim_ibus.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      - install -D -t /app/ pmim-ibus/README.md
+
+      # /app/bin
+      - install -Dm755 -t /app/bin/ run.sh
+      - install -Dm755 -t /app/bin/ pmim-ibus/flatpak/restart.sh
+
+      - install -Dm755 -t /app/bin/ dist-deno/deno
+
+      - "[ x$FLATPAK_ARCH == xx86_64 ] && install -Dm755 -t /app/bin/ dist-librush/release/ibrus || true"
+      - "[ x$FLATPAK_ARCH == xaarch64 ] && install -Dm755 -t /app/bin/ dist-librush/aarch64-unknown-linux-gnu/release/ibrus || true"
```
Comment (bbhtt) [inline io.github.fm_elpac.pmim_ibus.yml:50]: You can run this is:

```yaml
- |
  if [ "$FLATPAK_ARCH" = "aarch64" ]; then
    do something
  fi

**Comment by secext2022** [comment]
ok, updated

**Comment by secext2022** [comment]
bot, build io.github.fm_elpac.pmim_ibus

**Comment by secext2022** [comment]
bot, build io.github.fm_elpac.pmim_ibus

**Comment by secext2022** [comment]
Sorry, the previous build can not work, because flatpak unzip electron.zip does not preserve directory structure.
The new build fix it.

The new build is tested and work ok.
Sorry for my mistake.

**Comment by bbhtt** [comment]
It's stuck in this screen, is that how it is supposed to be?

image


Also it is downloading npm packages on first launch, will it be possible for you to include them here?

**Comment by secext2022** [comment]
@bbhtt   yes, this small window means this app work ok.

If you need to input Chinese, you need to install and configure ibus.

**Comment by secext2022** [comment]
The npm packages is downloaded by deno.

I use `deno vendor`, but it seems that currently deno can not vendor npm packages.

**Comment by bbhtt** [comment]
Ok seems fine then

**Pull request closed without merging**